src/useConditionalEffect.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, EffectCallback } from 'react';
+
+export const useConditionalEffect = (
+  condition: boolean,
+  effect: EffectCallback,
+  dependencies: any[]
```

Comment:
`DependencyList[]`?

Reply:
Does React export a `DependencyList` type? I thought I checked and saw that it's an array of any. I may be mistaken.

Reply:
Just plain `DependencyList | undefined` works